### PR TITLE
5549 keep po and pot files in sync

### LINKED ISSUE #5549: Translation update script should push updated templates before pulling translations
**Describe the bug**
It takes  2 rounds of `utils/merge_and_push_translations.sh` runs to properly propagate source string changes to `.po` files. `.pot` and `*.po` are inconsistent between the 2 runs.

**To Reproduce**
1. Check diff of `po/widelands/widelands.pot` in 9a4ed5d (4 Sep): strings `src/ui_fsmenu/addons/packager_box.cc:372`, and `:373` are removed, but still present in `po/widelands/<any lang>.po`.
2. Check diff of `po/widelands/<any lang>.po` in a1e798f (7 Sep): the above strings are only removed by this commit

**Expected behavior**
`.po` files should be updated at the same time as the `.pot`.

**Additional context**
I think the reason is that local updates of both `.pot` and `.po` files are handled by `utils/buildcat.py`, so solving this will need heavy refactoring (ideally merging?) of the 2 scripts.

### PR DESCRIPTION
**Pull request by SimonHeimberg**
**Type of change**
Bugfix / New feature / Refactoring / ...

**Issue(s) closed**
Fixes #5549

**How it works**
- push new translation template files to Transifex
- pull the updated translation files after

**Possible regressions**
Not sure if the updated translations are available immediately or there needs more waiting time. Could potentially fetch po files with different files. (I did not find any documentation about this.)

### CODE REVIEW

File: utils/merge_and_push_translations.sh [hunk 1]
```diff
@@ -57,7 +57,32 @@ echo "Working tree is clean, continuing"
...
+      git checkout $entry
+    fi
+  done
+  set -x
+}
+undo_oneliner_diffs
+
+if [ -z "$(git status -s)" ]; then
+  true # nothing changed, do not upload to transifex
+else
```
Comment (Noordfrees) [inline utils/merge_and_push_translations.sh]:
```suggestion
if [ -n "$(git status -s)" ]; then
  # Only upload to Transifex if anything changed
```

File: utils/merge_and_push_translations.sh [hunk 2]
```diff
@@ -57,7 +57,32 @@ echo "Working tree is clean, continuing"
...
+  # Push source catalogs to Transifex
+  tx push -s
+  sleep 5 # ?? # wait for translation files to be updated
```
Comment (Noordfrees) [inline utils/merge_and_push_translations.sh]: Transifex should be pretty quick with updating this, but maybe wait one minute to be sure. This workflow doesn't need to be very fast.

File: utils/merge_and_push_translations.sh [hunk 3]
```diff
@@ -97,15 +118,12 @@ python3 utils/fix_formatting.py --lua --dir data/i18n
...
-    git checkout $entry
-  fi
-done
-set -x
+undo_oneliner_diffs
+
+if [ -z "$(git status -s)" ]; then
+  echo "Run completed, nothing to commit."
+  exit 0
+fi
```
Comment (tothxa) [inline utils/merge_and_push_translations.sh:126]: This block shouldn't be moved. The next steps may also introduce changes from translations not in .po format.

Comment (SimonHeimberg) [reply utils/merge_and_push_translations.sh:126]: I do not understand, sorry.
The files are modified before. The next commands are "git add XXX" only and do not modify a file.
"git status -s" reports one line for every modified file, independent if the change is added (`M  fileName`) or only changed on disk (` M fileName`) or partially added (`MM fileName`). There is no filter for "*.po".

Comment (tothxa) [reply utils/merge_and_push_translations.sh:126]: Oops, sorry. You're right.

Comment (SimonHeimberg) [reply utils/merge_and_push_translations.sh:126]: No problem at all.

**Review by tothxa — APPROVED** [review]
I've looked more into `buildcat.py`, and indeed, `do_buildpo()`, `do_tunepo()` and `do_update_po()` are not actually used. I'm not sure whether they should be removed or just some comments added to this effect. Looks like they were disabled in 2010, so they've probably gone untested since then, so maybe removing them would be the right thing to do.

**Pull request merged**